src/generator/models/javascript.ts

```diff
@@ -99,6 +104,29 @@ export function prismaFieldToNexusType(field: DMMF.Field) {
   }
 }
 
+export function prismaFieldToNexusResolver(model: DMMF.Model, field: DMMF.Field) {
+  if (field.kind === 'object') {
+    return (root: any, _args: any, ctx: any) => {
+      const uniqueIdentifiers = resolveUniqueIdentifiers(model)
+      const missingIdentifiers = findMissingUniqueIdentifiers(root, uniqueIdentifiers)
+
+      if (missingIdentifiers !== null) {
+        throw new Error(
+          `Resolver ${model.name}.${
+            field.name
+          } is missing the following unique identifiers: ${missingIdentifiers.join(', ')}`
+        )
+      }
+
+      return ctx.prisma[lowerFirst(model.name)]
+        .findUnique({
```

Comment:
In my spec about relation resolvers https://gist.github.com/jasonkuhrt/fb5dcb58ba9bf68123460138cb17bc40#relationship-resolver I had imagined trying to use `findMany` with `where` as a general solution for all cases.

I'm ok shipping another solution for now if you're doing it, but just wanted to point this out at least.

The thing is, findUnique will not account for all relation cases.

But `findMany` with `where` could.

I realize `findUnique` might be more optimal, but I kind of want/wish to just punt that to Prisma lol, and let the "compiler" figure it out such that it optimizes to a `findUnique` automatically :D

I see NP using`findUnique` as an optimization. I'm not sure it makes a good starting place. If it is easier for you though, and makes the difference between shipping or not shipping, then ok.

Reply:
This solution comes from the old plugin too, i'm not sure if this is the more efficient but it solves all relation types without adding specific case to get the first item.

Reply:
How does `findUnique` work with `1:n` relations?

Reply:
here `findUnique` always retrieve the current root object. Then we let prisma client automatically retrieve related objects using the appropriate prisma client method.
So, Prisma client handle magically for us both 1-1 and 1-n :)

This is the logic implemented in the old plugin